src/edu/washington/escience/myria/operator/MergeJoin.java

```diff
@@ -453,9 +453,13 @@ private void leftAndRightEqual() throws Exception {
     // advance the one with the larger set of equal tuples because this produces fewer join tuples
     // not exact but good approximation
     final int leftSizeOfGroupOfEqualTuples =
-        leftRowIndex + TupleBatch.BATCH_SIZE * (leftBatches.size() - 1) - leftBeginIndex;
+        leftRowIndex
+            + TupleUtils.getBatchSize(generateSchema()) * (leftBatches.size() - 1)
```

Comment:
I think the schema should be the schema of the left batch, not the output of this join. They may have different schemas.